src/main/java/com/dev/eficiente/nosso/bolao/domain/service/UsuarioService.java

```diff
@@ -0,0 +1,35 @@
+package com.dev.eficiente.nosso.bolao.domain.service;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.dev.eficiente.nosso.bolao.domain.exception.NegocioException;
+import com.dev.eficiente.nosso.bolao.domain.model.Usuario;
+import com.dev.eficiente.nosso.bolao.domain.repository.UsuarioRepository;
+import com.dev.eficiente.nosso.bolao.infrastructure.security.CriptografiaUtil;
+
+@Service
+public class UsuarioService {
+
+	private final UsuarioRepository usuarioRepository;
+	private final CriptografiaUtil criptografiaUtil;
+	
+	@Autowired
+	public UsuarioService(UsuarioRepository usuarioRepository, CriptografiaUtil criptografiaUtil) {
+		this.usuarioRepository = usuarioRepository;
+		this.criptografiaUtil = criptografiaUtil;
+	}
+
+	@Transactional
+	public Usuario salvar(Usuario usuario) {
+		if(usuarioRepository.findByLogin(usuario.getLogin()).isPresent()) {

```

Comment:
Mas ainda não foi pedido essa verificação :). E isso é mais uma validação que a gente vai precisar fazer. Se é algo que pode acontecer, não lançamos exception :).

Reply:
Acabei fazendo coisa a mais aqui, depois que vi que era o próximo card.
Não entendi a parte:

> Se é algo que pode acontecer, não lançamos exception

Não é para lançar exceção agora, mas quando for necessário verificar terei que lançar certo?

Reply:
Opa, bom ponto. Vou até mandar audio sobre isso :). O ecossistema do Java já meio que reconheceu que a ideia de usar exceptions para indicar problemas de negócio foi por água abaixo. Pode ver que nenhum framework atualizado que você usa te força a tratar exception. Inclusive não existe nenhuma outra linguagem com essa característica :). 

Depois de tudo que eu já estudei + experimentação minha sugestão é: só lance exception para situações que a funcionalidade não previu que poderia acontecer. Se existir chance de alguma situação acontecer no sistema, devemos usar tipos convencionais para tratar isso. Exception é só para casos realmente inesperados :).